Application/Query Handlers/GetUserByIdQueryHandler.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using Application.Queries;
+using Domain.Entities;
+using MediatR;
+using Persistence.Repository_Interfaces;
+
+namespace Application.Query_Handlers
+{
+    public class GetUserByIdQueryHandler : IRequestHandler<GetUserByIdQuery, ApplicationUser>
+    {
+        private readonly IUserRepository _userRepository;
+
+        public GetUserByIdQueryHandler(IUserRepository userRepository)
+        {
+            _userRepository = userRepository;
+        }
+
+        public async Task<ApplicationUser> Handle(GetUserByIdQuery request,
+            CancellationToken cancellationToken)
+        {
+            var user = _userRepository.GetUserByIdAsync(request.Id);
+                
+            if (user == null) return null;
```

Comment:
Please check your code, you try to check nullable for Task object. IT IS SUPER USELESS
Just add await for 22 line of code

Reply:
Resolved